tcp_check/datadog_checks/tcp_check/tcp_check.py

```diff
@@ -54,25 +50,25 @@ def __init__(self, name, init_config, instances):
                 self.socket_type = socket.AF_INET
             except Exception:
                 msg = "URL: {} is not a correct IPv4, IPv6 or hostname".format(self.url)
-                raise BadConfException(msg)
+                raise ConfigurationError(msg)
 
     def check(self, instance):
         start = time.time()
         try:
-            self.log.debug("Connecting to {} {}".format(self.addr, self.port))
+            self.log.debug("Connecting to %s %d", self.addr, self.port)
             sock = socket.socket(self.socket_type)
             try:
                 sock.settimeout(self.timeout)
                 sock.connect((self.addr, self.port))
             finally:
                 sock.close()
 
+            self.log.debug("%s:%d is UP", self.addr, self.port)
+            self.report_as_service_check(AgentCheck.OK, 'UP')
```

Comment:
Would be preferable to put this in the `else` clause to limit the scope of the `try`.

```python

    def check(self, instance):
        start = time.time()
        try:
            self.log.debug("Connecting to {} {}".format(self.addr, self.port))
            sock = socket.socket(self.socket_type)
            try:
                sock.settimeout(self.timeout)
                sock.connect((self.addr, self.port))
            finally:
                sock.close()

        except socket.timeout as e:
            ...
        except socket.error as e:
            ...
        except Exception as e:
            ...
        else:
            self.log.debug("%s:%d is UP", self.addr, self.port)
            self.report_as_service_check(AgentCheck.OK, 'UP')
        
        ...

```